src/System.Device.Gpio/System/Device/Gpio/PinValue.cs

```diff
@@ -7,15 +7,45 @@ namespace System.Device.Gpio
     /// <summary>
     /// Represents a value for a pin.
     /// </summary>
-    public enum PinValue
+    public readonly struct PinValue : IEquatable<PinValue>
     {
+        // This isn't bool so the struct will retain blittable. This
+        // allows arrays of PinValue and structs that contain PinValue to
+        // be stack allocated.
+        private readonly byte _value;
+
+        private PinValue(byte value) => _value = value;
+
         /// <summary>
-        /// The value of the pin is low.
+        /// The value of the pin is high.
         /// </summary>
-        Low = 0,
+        public static PinValue High => new PinValue(1);
+
         /// <summary>
-        /// The value of the pin is high.
+        /// The value of the pin is low.
         /// </summary>
-        High = 1
+        public static PinValue Low => new PinValue(0);
+
+        public static implicit operator PinValue(int value) => value == 0 ? Low : High;
+        public static implicit operator PinValue(bool value) => value ? High : Low;
+        public static explicit operator byte(PinValue value) => value._value;
+        public static explicit operator int(PinValue value) => value._value;
+        public static explicit operator bool(PinValue value) => value._value == 0 ? false : true;
+
+        public bool Equals(PinValue other) => other._value == _value;
```

Comment:
I wonder if in here we should only really check if the PinValue is either high or low. I'm not sure if we should flag them as not equal when both are High except that one has a non-1 value.

Reply:
It is really hard to get a non standard value in here- you'd have to use unsafe code. I could do a more complicated check here but I thought it would be better to be as performant as possible and shame on you for using pointers to force the struct to -1. 😀